data_structures/linked_list/__init__.py

```diff
@@ -6,17 +6,22 @@ def __init__(self, item, next):
 class LinkedList:
     def __init__(self):
         self.head = None
+        self.size=0
 
     def add(self, item):
         self.head = Node(item, self.head)
+        self.size+=1
 
     def remove(self):
         if self.is_empty():
             return None
         else:
             item = self.head.item
             self.head = self.head.next
+            self.size-=1
             return item
 
     def is_empty(self):
         return self.head is None
+    def size(self):
```

Comment:
Please define [`__len__()`](https://docs.python.org/3/reference/datamodel.html#object.__len__) instead.

Is it better to maintain a `size` variable or is better to visit all nodes and count them?

Reply:
Sir I think maintaining size variable is better